src/box-container/types.ts

```diff
@@ -17,4 +17,5 @@ export interface BoxContainerProps {
     subComponentTestIds?: BoxContainerSubComponentTestIds | undefined;
     "data-testid"?: string | undefined;
     id?: string | undefined;
+    isHeaderContainerClickable?: boolean | undefined;
```

Comment:
trying to keep it succinct and follow the existing boolean naming convention 😵 

```suggestion
    clickableHeader?: boolean | undefined;
```